src/Slugger.js

```diff
@@ -6,28 +6,50 @@ module.exports = class Slugger {
     this.seen = {};
   }
 
-  /**
-   * Convert string to unique id
-   */
-  slug(value) {
-    let slug = value
+  serialize(value) {
+    return value
       .toLowerCase()
       .trim()
       // remove html tags
       .replace(/<[!\/a-z].*?>/ig, '')
       // remove unwanted chars
       .replace(/[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,./:;<=>?@[\]^`{|}~]/g, '')
       .replace(/\s/g, '-');
+  }
 
+  /**
+   * Finds the next safe (unique) slug to use
+   */
+  getNextSafeSlug(originalSlug, isDryRun) {
+    let slug = originalSlug;
+    let occurenceAccumulator = 0;
     if (this.seen.hasOwnProperty(slug)) {
-      const originalSlug = slug;
+      occurenceAccumulator = this.seen[originalSlug];
       do {
-        this.seen[originalSlug]++;
-        slug = originalSlug + '-' + this.seen[originalSlug];
+        occurenceAccumulator++;
+        slug = originalSlug + '-' + occurenceAccumulator;
       } while (this.seen.hasOwnProperty(slug));
     }
-    this.seen[slug] = 0;
-
+    if (!isDryRun) {
+      this.seen[originalSlug] = occurenceAccumulator;
+      this.seen[slug] = 0;
+    }
     return slug;
   }
+
+  /**
+   * Convert string to unique id
+   */
+  slug(value) {
+    const slug = this.serialize(value);
+    return this.getNextSafeSlug(slug);
+  }
+
+  /**
+   * Get slug text without incrementing accumulator
+   */
+  slugText(value) {
+    const slug = this.serialize(value);
+    return this.getNextSafeSlug(slug, true);
+  }
```

Comment:
Instead of having a new method could we just pass an option. That way in the future if we want to create more options we don't need to keep adding methods and changing parameters.

```suggestion
  slug(value, option = {}) {
    const slug = this.serialize(value);
    return this.getNextSafeSlug(slug, options.dryrun);
  }
```

Reply:
Sure, this sounds like a good idea, I'll make the change and update the tests!

Reply:
What's your preferred way of documenting the option for intellisense? JSDoc?